app/main.py

```diff
@@ -1,3 +1,57 @@
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: Union[int, float]) -> None:
+        self.km = km
+
+    def __add__(self, other: Union["Distance", int, float]) -> "Distance":
+        if isinstance(other, int | float):
```

Comment:
The use of `int | float` for type checking is incorrect for Python versions prior to 3.10. Use `isinstance(other, (int, float))` instead.